sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -161,6 +165,28 @@ public Mono<ShouldRetryResult> shouldRetry(Exception e) {
                 clientException);
         }
 
+        if (clientException != null && Exceptions.isStatusCode(clientException, HttpConstants.StatusCodes.REQUEST_TIMEOUT)) {
+            logger.info(
+                "Request timeout - IsReadRequest {}, IsWebExceptionRetriable {}, NonIdempotentWriteRetriesEnabled {}",
+                this.isReadRequest,
+                false,
+                this.request.getNonIdempotentWriteRetriesEnabled(),
+                e);
+
+            return this.shouldRetryOnRequestTimeout(
+                this.isReadRequest,
+                this.request.getNonIdempotentWriteRetriesEnabled()
+            );
+        }
+
+        if (clientException != null &&
+            Exceptions.isStatusCode(clientException, HttpConstants.StatusCodes.INTERNAL_SERVER_ERROR) &&
+            Exceptions.isSubStatusCode(clientException, HttpConstants.SubStatusCodes.UNKNOWN)) {
+
+            logger.info("Internal server error - IsReadRequest {}", this.isReadRequest, e);
```

Comment:
same logging question as above

Reply:
Switched to use `debug` mode here - enhance the diagnostics to have more relevant failure / success details for a partition.